Crossref REST.js

```diff
@@ -0,0 +1,1733 @@
+				let identifier;
+				let reviewOf = result.relation['is-review-of'][0];
+				let type = reviewOf['id-type'];
+				let id = reviewOf.id;
+				if (type === 'doi') {
+					identifier = '<a href="https://doi.org/' + id + '">https://doi.org/' + id + '</a>';
+				}
+				else if (type === 'url') {
+					identifier = '<a href=\"' + id + '\">' + id + '</a>';
+				}
+				else {
+					identifier = id;
+				}
+				item.notes.push('Review of ' + identifier);
+			}
+		}
+		else {
+			item = new Zotero.Item('document');
+		}
+
+		parseCreators(result, item, creatorTypeOverrideMap);
+
+		if (result.description) {
+			item.notes.push(result.description);
+		}
+
+		item.abstractNote = result.abstract && removeUnsupportedMarkup(result.abstract);
+		item.pages = result.page;
+		item.ISBN = result.ISBN && result.ISBN.join(', ');
+		item.ISSN = result.ISSN && result.ISSN.join(', ');
+		item.issue = result.issue;
+		item.volume = result.volume;
+		item.language = result.language;
+		item.edition = result['edition-number'];
+		item.university = item.institution = item.publisher = result.publisher;
+
+		if (result['container-title'] && result['container-title'][0]) {
+			if (['journalArticle'].includes(item.itemType)) {
+				item.publicationTitle = result['container-title'][0];
+			}
+			else if (['conferencePaper'].includes(item.itemType)) {
+				item.proceedingsTitle = result['container-title'][0];
+			}
+			else if (['book'].includes(item.itemType)) {
+				item.series = result['container-title'][0];
+			}
+			else if (['bookSection'].includes(item.itemType)) {
+				item.bookTitle = result['container-title'][0];
+			}
+			else {
+				item.seriesTitle = result['container-title'][0];
+			}
+		}
+
+		item.conferenceName = result.event && result.event.name;
+
+		// "short-container-title" often has the same value as "container-title", so it can be ignored
+		if (result['short-container-title'] && result['short-container-title'][0] !== result['container-title'][0]) {
+			item.journalAbbreviation = result['short-container-title'][0];
+		}
+
+		if (result.event && result.event.location) {
+			item.place = result.event.location;
+		}
+		else if (result.institution && result.institution[0] && result.institution[0].place) {
+			item.place = result.institution[0].place.join(', ');
+		}
+		else {
+			item.place = result['publisher-location'];
+		}
+
+		item.institution = item.university = result.institution && result.institution[0] && result.institution[0].name;
+
+		// Prefer print to other dates
+		if (parseDate(result['published-print'])) {
+			item.date = parseDate(result['published-print']);
+		}
+		else if (parseDate(result.issued)) {
+			item.date = parseDate(result.issued);
+		}
+
+		if (result.DOI) {
+			if (ZU.fieldIsValidForType('DOI', item.itemType)) {
+				item.DOI = result.DOI;
+			}
+			// Add DOI to extra for unsupported items
+			else {
+				item.extra = (item.extra ? item.extra + '\n' : '') + 'DOI: ' + result.DOI;
```

Comment:
We shouldn't need to do this manually. It should get set automatically if not valid for the field.

Reply:
(But that might not show up in tests.)

Reply:
Yes, it was exactly like that.